pkg/platform/kube/test/platform_test.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"encoding/base64"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube"
+	nuclioio "github.com/nuclio/nuclio/pkg/platform/kube/apis/nuclio.io/v1beta1"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DeployFunctionTestSuite struct {
+	KubeTestSuite
+}
+
+func (suite *DeployFunctionTestSuite) SetupSuite() {
+	suite.KubeTestSuite.SetupSuite()
+
+	// start controller in background
+	go suite.Controller.Start() // nolint: errcheck
+}
+
+func (suite *DeployFunctionTestSuite) TearDownSuite() {
+
+	// remove nuclio function leftovers
+	_, err := suite.executeKubectl([]string{"delete", "nucliofunctions", "--all"}, nil)
```

Comment:
I feel like this is something we might want in every test suite. Maybe move this to the base suite?

Reply:
Moved to `TearDownTest` to ensure no leftovers existing after _each_ test

+ moved to base